lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -279,6 +294,18 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 				continue
 			}
 
+			// If the user has manually defined private hop hints, ignore
+			// any channels not in the array.
+			if len(invoice.PrivateChannels) > 0 {
+				// Check to see whether the ShortChannelID exists in the private
+				// channel map. If it isn't, don't add this private channel to
+				// the list of hop hints.
+				sid := channel.ShortChannelID.ToUint64()
+				if _, ok := privateChanMap[sid]; !ok {
+					continue
```

Comment:
As is, users can provide public or already closed channels in this list and they will just be filtered out by this + the isPublic check, which could result in there being no hints added when the user gives bad input?

The extreme case would be to error if they provide hops and we end up using none, but maybe just add some logging along the lines of `private invoice: received {len} suggested channels, created: {numHints} hints`?

Reply:
Gotcha, agree